### PR TITLE
chronograf: init at 1.7.11

### PR DESCRIPTION
**Pull request by NefixEstrada**
###### Motivation for this change
Adding Chronograf to NixOS to have the whole TICK stack available. Also, I want to use it :P

###### Things done
- Created the chronograf package
- Created the chronograf module
- Added myself to the maintainers list

###### Requirements
- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/monitoring/chronograf.nix
Comment (aanderse) [inline nixos/modules/services/monitoring/chronograf.nix]: It would be nice if you could split this over multiple lines for easier reading.

Comment (aanderse) [inline nixos/modules/services/monitoring/chronograf.nix]: It appears that you have a reasonable set of defaults :+1: I would consider documenting how users can customize the install by taking advantage of [environment variables](https://docs.influxdata.com/chronograf/v1.7/administration/config-options). You might even consider making a `config` option which takes an attribute set and generates environment variables for the user, though if you go this route you would need special handling for sensitive options, such as passwords and whatnot.

Please let me know if you need any help on this.

**Review by aanderse — CHANGES_REQUESTED** [review]
Welcome to nixpkgs and thank you for contributing! I have left a few comments on the NixOS module I hope you will find helpful. I'm not a `yarn2nix` (or `nodejs` on nix in general) person, so I'll leave it to someone else to review that code.

Please let me know if you have any questions at all.

**Comment by NefixEstrada** [comment]
@aanderse Everything should be ready now! About the maintainers stuff, do you mean another PR?

**Comment by aanderse** [comment]
@NefixEstrada If you're familiar with rewriting git history I mean you should rewrite the history of this PR such that the change you made to `maintainers/maintainer-list.nix` should be a single commit, before the commit adding the package and service. The other 2 commits can be squashed into a single commit.

If you're not familiar with this git workflow please read up on git history rewriting and feel free to ask if you have any questions.

**Comment by NefixEstrada** [comment]
It seems that I also edited the 2 commits previous than the one mines. I'm not really sure how this happened. Sorry for adding everyone to the conversation.

Other than that, I separated the maintainers commit and squashed the other to into a single commit. Should I try to do the same without modifying those commits? If so, could someone give me a hand? Thanks!


<details>
<summary>(Commands I've executed)</summary>

```sh
git rebase -i HEAD~2
git reset HEAD^
git add maintainers/maintainer-list.nix
git commit
git add .
git commit
git rebase --continue
git push --force
```
</details>

File: nixos/modules/services/monitoring/chronograf.nix
Comment (Mic92) [inline nixos/modules/services/monitoring/chronograf.nix]: Does this match the upstream default?

File: nixos/modules/services/monitoring/chronograf.nix
Comment (Mic92) [inline nixos/modules/services/monitoring/chronograf.nix]: You can use in all three cases `optionalString` in case the variables are empty.

File: pkgs/development/compilers/rust/rustc.nix
Comment (Mic92) [inline pkgs/development/compilers/rust/rustc.nix]: The rust upgrade seems unrelated.

File: pkgs/servers/monitoring/chronograf/default.nix
Comment (Mic92) [inline pkgs/servers/monitoring/chronograf/default.nix]: I think yarn2nix does not work in the mode nixpkgs evaluates.

**Comment by Mic92** [comment]
+10000 lines of yarn2nix is quite a lot for nixpkgs. Is there a way to integrate this into `pkgs/development/node-packages/node-packages-v10.json`?

**Comment by aanderse** [comment]
@NefixEstrada are you motivated to continue with this PR?

**Comment by NefixEstrada** [comment]
If someone else wants to continue with the PR, feel free to do it. I feel that my Nix skills level might be too low for correctly add the service.

If you tell me how should I solve the issues that you've pointed out, I can finish the work, but otherwise I'm not going to be able to finish it :(

**Comment by aanderse** [comment]
@NefixEstrada which issues are you specifically having trouble with? The issues I raised seem to be resolved by you for the most part, with the exception of git history. If you're specific on what you're having issues with maybe someone can help you out.

**Comment by NefixEstrada** [comment]
It's both the git history and the fact that the `package.json` shouldn't be in nixpkgs

**Comment by Mic92** [comment]
For grafana we ended up taking the javascript assets from the binary package and only building the go project itself.

**Comment by sorpaas** [comment]
For anyone who wants to use chronograf in the mean time, a docker version is simple to write:

````
  docker-containers."chronograf" = {
    after = [ "influxdb.service" ];
    args = [
      "--network host"
      "-v /var/lib/chronograf:/var/lib/chronograf"
      "--user ${chronografUid}:${chronografGid}"
      "chronograf"
    ];
    dockerNamePrefix = false;
  };
```

I also found https://github.com/m1cr0man/nix-configs/tree/9520cc569f690d0f3603be91c637cef23d00ffff/packages/chronograf but that does not seem to build, unfortunately.

File: pkgs/servers/monitoring/chronograf/default.nix
Comment (jtojnar) [inline pkgs/servers/monitoring/chronograf/default.nix]: Use proper list for compatibility with #72074 

```suggestion
  buildFlags = [
    "-ldflags="
    "-s"
    "-X" "main.version=${version}"
  ];
```

File: pkgs/servers/monitoring/chronograf/default.nix
Comment (jtojnar) [inline pkgs/servers/monitoring/chronograf/default.nix]: Does not `dontInstall = true` work?

File: nixos/modules/services/monitoring/chronograf.nix
Comment (jtojnar) [inline nixos/modules/services/monitoring/chronograf.nix]: Is it possible to reuse the [upstream service file](https://github.com/influxdata/chronograf/blob/0568cac0406fe8c15320f9864e2102635aff2cf0/etc/scripts/chronograf.service) after patching it?

**Comment by Mindavi** [comment]
Closing. Can be reopened or preferably, resubmitted if someone still finds this useful.

**Pull request closed without merging**
